File: android/core/domain/src/main/java/com/enterprise/core/domain/usecases/ShareUseCase.kt
package com.enterprise.core.domain.usecases

import com.enterprise.core.domain.interfaces.ShareService
import javax.inject.Inject

/** Shares content via native OS share sheets (WhatsApp, iMessage, etc.). */
class ShareUseCase @Inject constructor(
    private val shareService: ShareService
) {
    suspend fun shareText(text: String, subject: String? = null) {
        shareService.shareText(text, subject)
    }

    suspend fun shareUrl(url: String, title: String? = null) {
        shareService.shareUrl(url, title)
    }
}

File: ios/EnterpriseMobileApp/Sources/Core/Domain/UseCases/ShareUseCase.swift
import Foundation

/// Use case for sharing content via native OS share sheets.
public struct ShareUseCase: Sendable {
    private let shareService: ShareService

    public init(shareService: ShareService) {
        self.shareService = shareService
    }

    public func share(text: String, url: URL? = nil) async {
        await shareService.share(text: text, url: url)
    }

    public func shareToWhatsApp(text: String) async -> Bool {
        await shareService.shareToWhatsApp(text: text)
    }
}
